bwrap-oci.c

```diff
@@ -0,0 +1,500 @@
+      if (terminal)
+        collect_options (con, "--dev-bind", "/dev/tty", "/dev/tty", NULL);
+    }
+  if (json_object_has_member (root, "cwd"))
+    {
+      JsonNode *cwd = json_object_get_member (root, "cwd");
+      collect_options (con, "--chdir", json_node_get_string (cwd), NULL);
+    }
+  if (json_object_has_member (root, "env"))
+    {
+      GList *members;
+      GList *iter;
+      members = json_array_get_elements (json_node_get_array (json_object_get_member (root, "env")));
+      for (iter = members; iter; iter = iter->next)
+        {
+          GVariant *env = json_gvariant_deserialize (iter->data, "s", NULL);
+          char *val = g_variant_dup_string (env, NULL);
+          gchar *sep = g_strrstr (val, "=");
+          if (!sep)
+            error (EXIT_FAILURE, 0, "invalid env setting\n");
+          *sep = '\0';
+          collect_options (con, "--setenv", val, sep + 1, NULL);
+          g_free (val);
+          g_variant_unref (env);
+        }
+    }
+  if (json_object_has_member (root, "selinuxLabel"))
+    {
+      JsonNode *label = json_object_get_member (root, "selinuxLabel");
+      collect_options (con, "--exec-label", json_node_get_string (label), NULL);
+    }
+  if (json_object_has_member (root, "user"))
+    {
+      JsonNode *user = json_object_get_member (root, "user");
+      JsonObject *userobj = json_node_get_object (user);
+      if (json_object_has_member (userobj, "uid"))
+        {
+          gint64 uid = json_node_get_int (json_object_get_member (userobj, "uid"));
+          gchar *argument = g_strdup_printf ("%" G_GINT64_FORMAT, uid);
+          collect_options (con, "--uid", argument, NULL);
+          g_free (argument);
+        }
+      if (json_object_has_member (userobj, "gid"))
+        {
+          gint64 gid = json_node_get_int (json_object_get_member (userobj, "gid"));
+          gchar *argument = g_strdup_printf ("%" G_GINT64_FORMAT, gid);
+          collect_options (con, "--gid", argument, NULL);
+          g_free (argument);
+        }
+    }
+  if (json_object_has_member (root, "args"))
+    {
+      GList *members;
+      GList *iter;
+      members = json_array_get_elements (json_node_get_array (json_object_get_member (root, "args")));
+      for (iter = members; iter; iter = iter->next)
+        {
+          GVariant *arg = json_gvariant_deserialize (iter->data, "s", NULL);
+          const char *val = g_variant_get_string (arg, NULL);
+          collect_args (con, val, NULL);
+        }
+    }
+}
+
+static void
+dump_argv (char **argv)
+{
+  gboolean first = TRUE;
+  while (*argv)
+    {
+      g_print ("%s%s", first ? "" : " ", *argv);
+      first = FALSE;
+      argv++;
+    }
+  g_print ("\n");
+}
+
+static void
+generate_seccomp_rules_file (struct context *context)
+{
+  if (context->seccomp)
+    {
+      char fdstr[10];
+      int fd = open (".", O_TMPFILE | O_RDWR, S_IRUSR | S_IWUSR);
```

Comment:
the temporary file will be used by bubblewrap to read the seccomp rules:

`bwrap ... --seccomp FD ...`

so we need to keep it open in order to be used by bubblewrap